### PR TITLE
Issue #15340: created inputformatted file for section 4.8.2.2 declared when needed

### PR DESCRIPTION
**Pull request by Zopsss**
#15340

### CODE REVIEW

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule4822declaredwhenneeded/InputFormattedDeclaredWhenNeeded.java [hunk 1]
```diff
@@ -0,0 +1,752 @@
...
+    int c = 0;
+    int a = 3;
+    int b = 2;
+    { // violation ''block lcurly' has incorrect indentation level 4, expected level should be 6.'
+      a = a + b;
+      c = b;
+    } // violation ''block rcurly' has incorrect indentation level 4, expected level should be 6.'
+    { // violation ''block lcurly' has incorrect indentation level 4, expected level should be 6.'
+      c--;
+    } // violation ''block rcurly' has incorrect indentation level 4, expected level should be 6.'
```
Comment (Zopsss) [inline src/it/resources/com/google/checkstyle/test/chapter4formatting/rule4822declaredwhenneeded/InputFormattedDeclaredWhenNeeded.java:30]: these are false positives from our side and it is a known issue https://github.com/checkstyle/checkstyle/issues/14294

**Review by rdiachenko — APPROVED** [review]
lgtm

**Review by romani — CHANGES_REQUESTED** [review]
There is diff between files around `new ActionListener()`  and AuthUpdateTask please fix.

Indentation issue, should be signed back to Input and have no diff.

**Comment by Zopsss** [comment]
> There is diff between files around `new ActionListener()` and AuthUpdateTask please fix.
> 
> Indentation issue, should be signed back to Input and have no diff.

```diff
$ diff -u src/it/resources/com/google/checkstyle/test/chapter4formatting/rule4822declaredwhenneeded/InputDeclaredWhenNeeded.java src/it/resources/com/google/checkstyle/test/chapter4formatting/rule4822declaredwhenneeded/InputFormattedDeclaredWhenNeeded.java
--- src/it/resources/com/google/checkstyle/test/chapter4formatting/rule4822declaredwhenneeded/InputDeclaredWhenNeeded.java      2024-09-01 18:38:14.841648900 +0530
+++ src/it/resources/com/google/checkstyle/test/chapter4formatting/rule4822declaredwhenneeded/InputFormattedDeclaredWhenNeeded.java     2024-09-01 18:37:11.213480100 +0530
@@ -5,29 +5,29 @@
 import java.util.TimeZone;
 
 /** Some javadoc. */
-public class InputDeclaredWhenNeeded {
+public class InputFormattedDeclaredWhenNeeded {
 
   private static int test1 = 0;

   static {
     int b = 0;
     int d = 0;
-    {
+    { // violation ''block lcurly' has incorrect indentation level 4, expected level should be 6.'
       d = ++b;
-    }
+    } // violation ''block rcurly' has incorrect indentation level 4, expected level should be 6.'
   }

   static {
     int c = 0;
     int a = 3;
     int b = 2;
-    {
+    { // violation ''block lcurly' has incorrect indentation level 4, expected level should be 6.'
       a = a + b;
       c = b;
-    }
-    {
+    } // violation ''block rcurly' has incorrect indentation level 4, expected level should be 6.'
+    { // violation ''block lcurly' has incorrect indentation level 4, expected level should be 6.'
       c--;
-    }
+    } // violation ''block rcurly' has incorrect indentation level 4, expected level should be 6.'
     a = 7;
   }

@@ -40,7 +40,7 @@
   }

   /** Some javadoc. */
-  public InputDeclaredWhenNeeded(int test1) {
+  public InputFormattedDeclaredWhenNeeded(int test1) {
     int temp = -1;
     this.test1 = test1;
     temp = test1; // DECLARATION OF VARIABLE 'temp' SHOULD BE HERE (distance = 2)
@@ -49,7 +49,7 @@
   /** Some javadoc. */
   public boolean testMethod() {
     int temp = 7;
-    new InputDeclaredWhenNeeded(2);
+    new InputFormattedDeclaredWhenNeeded(2);
     String.valueOf(temp); // DECLARATION OF VARIABLE 'temp' SHOULD BE HERE (distance = 2)
     boolean result = false;
     String str = "";
@@ -65,10 +65,10 @@
     int count;
     int a = 3;
     int b = 2;
-    {
+    { // violation ''block lcurly' has incorrect indentation level 4, expected level should be 6.'
       a = a + b - 5 + 2 * a;
       count = b; // DECLARATION OF VARIABLE 'count' SHOULD BE HERE (distance = 2)
-    }
+    } // violation ''block rcurly' has incorrect indentation level 4, expected level should be 6.'
   }

   /** Some javadoc. */
@@ -156,15 +156,15 @@
     int c = 0;
     int m = 0;
     int n = 0;
-    {
+    { // violation ''block lcurly' has incorrect indentation level 4, expected level should be 6.'
       c++;
       b++;
-    }
-    {
+    } // violation ''block rcurly' has incorrect indentation level 4, expected level should be 6.'
+    { // violation ''block lcurly' has incorrect indentation level 4, expected level should be 6.'
       n++; // DECLARATION OF VARIABLE 'n' SHOULD BE HERE (distance = 2)
       m++; // DECLARATION OF VARIABLE 'm' SHOULD BE HERE (distance = 3)
       b++;
-    }
+    } // violation ''block rcurly' has incorrect indentation level 4, expected level should be 6.'
   }

   /** Some javadoc. */
```

there only some extra false positive violations and some indentation changes diff between OG file and formatted file, should we just delete the formatted file and exclude this section from google-java-format.sh ?

https://github.com/checkstyle/checkstyle/blob/fec0f2829c16b5ea276d9a569d0174d4f8e68469/.ci/google-java-format.sh#L51

**Comment by romani** [comment]
yes, just format InputDeclaredWhenNeeded and put extra violations into it, I hope they such extra violations will be fixed soon.

**Comment by Zopsss** [comment]
done

**Review by romani — APPROVED** [review]
Ok to merge

**Pull request merged**